components/iis/component.go

```diff
@@ -0,0 +1,148 @@
+// Package iis contains code for the IIS component
+package iis
+
+import (
+	"fmt"
+	"github.com/DataDog/test-infra-definitions/common"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/components/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumiverse/pulumi-time/sdk/go/time"
+)
+
+// Output is an object that models the output of the resource creation
+// from the Component.
+// See https://www.pulumi.com/docs/concepts/resources/components/#registering-component-outputs
+type Output struct {
+	components.JSONImporter
+}
+
+// Component represent an IIServer and its sites.
+// See https://www.pulumi.com/docs/concepts/resources/components/
+type Component struct {
+	pulumi.ResourceState
+	components.Component
+	namer namer.Namer
+	host  *remote.Host
+}
+
+// Export registers a key and value pair with the current context's stack.
+func (c *Component) Export(ctx *pulumi.Context, out *Output) error {
+	return components.Export(ctx, c, out)
+}
+
+// NewServer creates a new IIS server component.
+// Example usage:
+//
+//	iisServer, err := iis.NewServer(ctx, awsEnv.CommonEnvironment, vm,
+//		iis.WithSite(iis.SiteDefinition{
+//			Name:            "TestSite1",
+//			BindingPort:     "*:8081:",
+//			SourceAssetsDir: srcDir,
+//		}),
+//	)
+//	if err != nil {
+//		return err
+//	}
+//	err = iisServer.Export(ctx, env.IISServer)
+func NewServer(ctx *pulumi.Context, e *config.CommonEnvironment, host *remote.Host, options ...Option) (*Component, error) {
+	if host.OS.Descriptor().Family() != os.WindowsFamily {
+		// Print flavor in case OS family don't match, as it's more precise than the family (and the .String() conversion already exists).
+		return nil, fmt.Errorf("wrong Operating System family, expected Windows, got %s", host.OS.Descriptor().Flavor.String())
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**